lib/cdo/redshift.rb

```diff
@@ -16,6 +41,54 @@ def initialize
   end
 
   def exec(sql_query)
-    @conn.exec(sql_query)
+    result = @conn.exec(sql_query)
+    # result_status is an undocumented method that returns the integer query status.
+    # res_status is an undocumented method that returns the string identifier for a given integer query status.
+    result_status = result.res_status(result.result_status)
+    raise PostgreSQLQueryError.new(result_status) unless QUERY_SUCCESS_CODES.include? result_status
+    result
+  end
+
+  # DMS Replication Tasks import data into temporary / staging tables prefixed with "_import_".
+  # Complete import of staged data from production database into Redshift by truncating each
+  # target table, moving all of the rows from each import table to its corresponding target table, and then dropping
+  # each temporary / staging import table.
+  def complete_table_import
+    Cdo::DMS::REDSHIFT_SCHEMAS_IMPORTED_FROM_DATABASE.each do |schema|
+      temporary_import_tables(schema).each do |import_table|
+        target_table = import_table.partition(TEMP_TABLE_PREFIX).last
+        truncate_table(schema, target_table)
+        move_rows(schema, import_table, schema, target_table)
+        drop_table(schema, import_table)
+      end
+    end
+  end
+
+  # Get names of tables in a Redshift schema that were loaded via DMS to stage imported data from the database.
+  # @param schema [String] The Redshift schema to search for database import staging tables.
+  # Returns Array of Redshift table names.
+  def temporary_import_tables(schema)
+    query = <<-SQL
+SET search_path TO #{schema};
+
+SELECT DISTINCT t.tablename
+FROM pg_table_def t LEFT JOIN pg_indexes i ON t.tablename = i.indexname
+WHERE  i.indexname IS NULL -- Don't count primary keys, which appear in pg_table_def, as a table.
+  AND  t.schemaname ='#{schema}'
+  AND t.tablename LIKE '#{TEMP_TABLE_PREFIX}%';
+SQL
+    exec(query).map {|row| row['tablename']}
+  end
```

Comment:
I'd prefer if the more complex DMS-specific methods here (`complete_table_import` and `temporary_import_tables`) were moved to another module specific to the DMS import behavior. (`RedshiftClient` is described as 'A thin wrapper around PG, providing a mechanism to execute SQL commands on our AWS Redshift instance.', and I'm concerned about muddying this class by straying too far from that description.)

Reply:
Good point.  The truncate/move/drop methods probably don't belong here either.

Reply:
I've moved these out to `RedshiftImport`